discussment-core/src/main/java/org/danekja/discussment/core/service/imp/DefaultPostService.java

```diff
@@ -64,6 +64,7 @@ public Post getPostById(long postId) throws AccessDeniedException {
 
     public Post sendReply(Post reply, Post post) throws AccessDeniedException {
         reply.setPost(post);
+        reply.setLevel(post.getLevel() + 1);
```

Comment:
Actually correct place for this would be then extra method Post.setAsReply(Post parent) which would call setPost and setLevel.

The Post entity needs to own its state.

Reply:
True. Where is a good place to test it? Integration test?